packages/expo-router/src/global-state/getRoutes.ts

```diff
@@ -0,0 +1,420 @@
+import { Platform } from 'react-native';
+
+import { DynamicConvention, RouteNode } from '../Route';
+import {
+  matchDeepDynamicRouteName,
+  matchDynamicName,
+  matchGroupName,
+  removeSupportedExtensions,
+} from '../matchers';
+import { RequireContext } from '../types';
+
+export type Options = {
+  ignore?: RegExp[];
+  preserveApiRoutes?: boolean;
+  ignoreRequireErrors?: boolean;
+  ignoreEntryPoints?: boolean;
+  unstable_platformExtensions?: boolean;
+  unstable_stripLoadRoute?: boolean;
+  unstable_alwaysIncludeSitemap?: boolean;
+  unstable_improvedErrorMessages?: boolean;
+};
+
+type DirectoryNode = {
+  layout?: RouteNode[];
+  views: Map<string, RouteNode[]>;
+  subdirectories: Map<string, DirectoryNode>;
+};
+
+const validPlatforms = new Set(['android', 'ios', 'windows', 'osx', 'native', 'web']);
```

Comment:
Need to add other out of tree platforms (tv?)

Reply:
Should just have ios, android, web, and native.